### PR TITLE
server: don't export B&R APIs if feature is not enabled globally

### PR DESCRIPTION
**Pull request by rhtyd**
This change will ensure that B&R APIs are not exported if the feature is not enabled globally. Make it non-dynamic to require restarting of mgmt server which is needed for the background sync task to be setup or not.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

File: api/src/main/java/org/apache/cloudstack/backup/BackupManager.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
     ConfigKey<Boolean> BackupFrameworkEnabled = new ConfigKey<>("Advanced", Boolean.class,
             "backup.framework.enabled",
             "false",
-            "Is backup and recovery framework enabled.", true, ConfigKey.Scope.Zone);
+            "Is backup and recovery framework enabled. Restart management server on global value change.", true, ConfigKey.Scope.Zone);
```
Comment (GabrielBrascher) [inline api/src/main/java/org/apache/cloudstack/backup/BackupManager.java:41]: According to the description, ADMIN needs to restart the management server and therefore this is not a dynamic setting. 

What do you think of changing the `isDynamic` param to `false`?

_ConfigKey constructor:_
```
public ConfigKey(String category, Class<T> type, String name, String defaultValue, String description, boolean isDynamic, Scope scope) {
    this(type, name, category, defaultValue, description, isDynamic, scope, null);
}
```

Comment (rhtyd) [reply api/src/main/java/org/apache/cloudstack/backup/BackupManager.java:41]: I'll change that.

Comment (rhtyd) [reply api/src/main/java/org/apache/cloudstack/backup/BackupManager.java]: Fixed thanks

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@davidjumani please review this, thnx

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
Tested on KVM env, it skips tests for dummy provider now:
```
Import provider backup offering from Dummy Backup and Recovery Provider ... SKIP: Skipping test cases which must only run for Simulator
Test VM backup lifecycle ... SKIP: Skipping test cases which must only run for Simulator

----------------------------------------------------------------------
Ran 2 tests in 0.124s

OK (SKIP=2)

```

**Pull request merged**